### PR TITLE
[js-yaml] fix: add missing listener option

### PR DESCRIPTION
**Pull request by jaulz**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: 
https://github.com/nodeca/js-yaml/blob/master/test/issues/0248-listener.js
https://github.com/nodeca/js-yaml/blob/master/lib/js-yaml/loader.js
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by jaulz** [comment]
Please note that the `State` type is superficial and does not expose everything yet. I don't think that we need to expose and define each and every detail since it's anyway rarely used.

File: types/js-yaml/index.d.ts [hunk 1]
```diff
@@ -48,6 +48,28 @@ export interface LoadOptions {
 	schema?: SchemaDefinition;
 	/** compatibility with JSON.parse behaviour. */
 	json?: boolean;
+	/** listener for parse events */
+	listener?: (eventType: EventType, state: State) => void;
```
Comment (ExE-Boss) [inline types/js-yaml/index.d.ts]: `listener` is called with a `this` context of type `State`:
```suggestion
	listener?(this: State, eventType: EventType, state: State): void;
```

File: types/js-yaml/js-yaml-tests.ts [hunk 2]
```diff
@@ -68,6 +68,12 @@ loadOpts = {
 loadOpts = {
 	schema: yaml.DEFAULT_SAFE_SCHEMA,
 };
+loadOpts = {
+	listener: (eventType, state) => {
+        state.position;
+        state.result;
+    },
+};
```
Comment (ExE-Boss) [inline types/js-yaml/js-yaml-tests.ts]:
```suggestion
loadOpts = {
	listener(eventType: yaml.EventType, state) {
		this; // $ExpectType State
		state; // $ExpectType State
		state.position;
		state.result;
	},
};
```

**Review by ExE-Boss — CHANGES_REQUESTED** [review]

**Comment by jaulz** [comment]
@ExE-Boss good catch, thanks!

**Review by ExE-Boss — APPROVED** [review]

**Pull request merged**